src/Rest/Resource.php

```diff
@@ -238,7 +187,19 @@ public function offsetGet($offset)
      */
     public function offsetSet($offset, $value)
     {
-        $this->data[$offset] = $value;
+        if (method_exists($this, 'set' . $offset)) {
+            return call_user_func([$this, 'set' . $offset], $value);
+        }
+
+        if (method_exists($this, 'get' . $offset)) {
+            throw new DomainException(
+                sprintf('Trying to set the read-only property "%s::%s"', get_class($this), $offset)
```

Comment:
`__CLASS__` and `get_class` have different use cases. We need `get_class` because we need class name of instance and not class name where method was written.